Python/ibmcloudsql/cos.py

```diff
@@ -0,0 +1,659 @@
+            "sjc-us-geo": "s3-api.sjc-us-geo.objectstorage.softlayer.net",
+            "eu-geo": "s3.eu-geo.objectstorage.softlayer.net",
+            "eu": "s3.eu-geo.objectstorage.softlayer.net",
+            "ams-eu-geo": "s3.ams-eu-geo.objectstorage.softlayer.net",
+            "fra-eu-geo": "s3.fra-eu-geo.objectstorage.softlayer.net",
+            "mil-eu-geo": "s3.mil-eu-geo.objectstorage.softlayer.net",
+            "us-south": "s3.us-south.objectstorage.softlayer.net",
+            "us-east": "s3.us-east.objectstorage.softlayer.net",
+            "jp-tok": "s3.jp-tok.objectstorage.softlayer.net",
+            "ap-geo": "s3.ap-geo.objectstorage.softlayer.net",
+            "ap": "s3.ap-geo.objectstorage.softlayer.net",
+            "tok-ap-geo": "s3.tok-ap-geo.objectstorage.softlayer.net",
+            "seo-ap-geo": "s3.seo-ap-geo.objectstorage.softlayer.net",
+            "hkg-ap-geo": "s3.hkg-ap-geo.objectstorage.softlayer.net",
+            "eu-de": "s3.eu-de.objectstorage.softlayer.net",
+            "eu-gb": "s3.eu-gb.objectstorage.softlayer.net",
+            "ams03": "s3.ams03.objectstorage.softlayer.net",
+            "che01": "s3.che01.objectstorage.softlayer.net",
+            "mel01": "s3.mel01.objectstorage.softlayer.net",
+            "tor01": "s3.tor01.objectstorage.softlayer.net",
+            "mon01": "s3.mon01.objectstorage.softlayer.net",
+            "osl01": "s3.osl01.objectstorage.softlayer.net",
+            "sao01": "s3.sao01.objectstorage.softlayer.net",
+            "seo01": "s3.seo01.objectstorage.softlayer.net"
+        }
+        self.endpoint = None
+        self.bucket = None
+        self.prefix = None
+
+    def get_exact_url(self, cos_url):
+        """convert COS URL from using alias to exact URL"""
+        key = cos_url.split("/")[2]
+        x = cos_url.replace(key, self.endpoint_alias_mapping.get(key, key), 1)
+        return x
+
+    def get_endpoint(self, cos_url):
+        self.endpoint = cos_url.split("/")[2]
+        self.endpoint = self.endpoint_alias_mapping.get(
+            self.endpoint, self.endpoint)
+        return self.endpoint
+
+    def get_bucket(self, cos_url):
+        self.bucket = cos_url.split("/")[3]
+        return self.bucket
+
+    def get_prefix(self, cos_url):
+        self.prefix = cos_url[cos_url.replace('/', 'X', 3).find('/') + 1:]
+        if len(self.prefix) > 0 and self.prefix[-1] == '*':
+            self.prefix = self.prefix[:-1]
+            self.fourth_slash = cos_url.replace('/', 'X', 3).find('/')
+        return self.prefix
+
+    def analyze_cos_url(self, cos_url):
+        """ return a namedtuple containing the 3 fields
+
+        * bucket
+        * endpoint
+        * prefix
+        """
+        nt = namedtuple("COSURL", "endpoint bucket prefix")
+        mycontainer = nt(self.get_endpoint(cos_url), self.get_bucket(cos_url),
+                         self.get_prefix(cos_url))
+        return mycontainer
+
+
+# Create a child class that inherits from Parent
```

Comment:
Please provide a semantic description of the class

Reply:
Done in commit `4c3eb6593e7ab5a1d2f9f0cfe3ac084cc1ac0b68`